From: srinivas pulukuru <srinivas.pulukuru@ti.com>
Date: Fri, 6 May 2011 12:45:11 +0000 (-0500)
Subject: OMAP4 DSS HDMI: fix kernel crash due to incorrect disabling of digital channel
X-Git-Url: http://omapzoom.org/?p=kernel%2Fomap.git;a=commitdiff_plain;h=518d3c859b78f5f2c8e32caf8c4641c9462d8f41

OMAP4 DSS HDMI: fix kernel crash due to incorrect disabling of digital channel

1. HDMI_WP_VIDEO_CFG.ENABLE[31] bit should be reset first and SW
should wait for the VIDEO_FRAME_DONE before resetting the
digital channel. The digital channel was being disabled before
stoping the HDMI video frames. This was causing green frames on TV during
suspend and some times resulting in kernel crash.

Change-Id: I434670876fe4990040900c616000cd10d0c900df
Signed-off-by: srinivas pulukuru <srinivas.pulukuru@ti.com>
---

diff --git a/drivers/video/omap2/dss/hdmi.c b/drivers/video/omap2/dss/hdmi.c
index 3946ced91dee9cdca62578443eb545cbd19b9b0c..3008213 100644
--- a/drivers/video/omap2/dss/hdmi.c
+++ b/drivers/video/omap2/dss/hdmi.c
@@ -1641,7 +1641,7 @@ static void hdmi_power_off_phy(struct omap_dss_device *dssdev)
 	HDMI_W1_StopVideoFrame(HDMI_WP);
 	hdmi_set_irqs(1);
 
-	dispc_enable_digit_out(0);
+	dssdev->manager->disable(dssdev->manager);
 
 	if (hdmi.hdmi_stop_frame_cb)
 		(*hdmi.hdmi_stop_frame_cb)();
@@ -1753,9 +1753,8 @@ static int hdmi_set_power(struct omap_dss_device *dssdev,
 		} else if (power_need == HDMI_POWER_MIN) {
 			r = hdmi_min_enable();
 		} else {
-                        dssdev->manager->disable(dssdev->manager);
-			omap_dss_stop_device(dssdev);
 			hdmi_power_off(dssdev);
+			omap_dss_stop_device(dssdev);
 		}
 		if (!r)
 			hdmi_power = power_need;